independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/BootstrapAppModelResolver.java

```diff
@@ -297,37 +352,64 @@ private VersionRangeResult resolveVersionRangeResult(AppArtifact appArtifact, St
 
     static List<AppDependency> toAppDepList(DependencyNode rootNode) {
         final List<DependencyNode> depNodes = rootNode.getChildren();
-        if(depNodes.isEmpty()) {
+        if (depNodes.isEmpty()) {
             return Collections.emptyList();
         }
-        final List<AppDependency> appDeps =  new ArrayList<>();
+        final List<AppDependency> appDeps = new ArrayList<>();
         collect(depNodes, appDeps);
         return appDeps;
     }
 
     private static void collect(List<DependencyNode> nodes, List<AppDependency> appDeps) {
-        for(DependencyNode node : nodes) {
+        for (DependencyNode node : nodes) {
             collect(node.getChildren(), appDeps);
             final Dependency dep = node.getDependency();
             appDeps.add(new AppDependency(toAppArtifact(node.getArtifact()), dep.getScope(), dep.isOptional()));
         }
     }
 
     private static Artifact toAetherArtifact(AppArtifact artifact) {
-        return new DefaultArtifact(artifact.getGroupId(), artifact.getArtifactId(), artifact.getClassifier(), artifact.getType(), artifact.getVersion());
+        Artifact defaultArtifact = new DefaultArtifact(artifact.getGroupId(), artifact.getArtifactId(),
+                artifact.getClassifier(),
+                artifact.getType(), artifact.getVersion());
+        if (artifact.getPath() != null) {
+            defaultArtifact = defaultArtifact.setFile(artifact.getPath().toFile());
+        }
+        return defaultArtifact;
     }
 
     private static AppArtifact toAppArtifact(Artifact artifact) {
-        final AppArtifact appArtifact = new AppArtifact(artifact.getGroupId(), artifact.getArtifactId(), artifact.getClassifier(), artifact.getExtension(), artifact.getVersion());
+        final AppArtifact appArtifact = new AppArtifact(artifact.getGroupId(), artifact.getArtifactId(),
+                artifact.getClassifier(), artifact.getExtension(), artifact.getVersion());
         final File file = artifact.getFile();
-        if(file != null) {
+        if (file != null) {
             appArtifact.setPath(file.toPath());
         }
         return appArtifact;
     }
 
+    private static AppArtifact toAppArtifact(Artifact artifact, Map<AppKey, AppArtifact> versions) {
```

Comment:
Ah, this method is actually not used. Is it? So AppKey can simply be removed with this method.